Latest failure: dbus-message.c:4372:36: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]

And of course I had to eat my words.  Looks like one part of the patch was actually necessary...  Apparently the sign mismatch problem is not triggered in a 64 bit build.  Patch needs to be refreshed if we keep it.

Oh and I guess the "blacklisting" of the hardening flags worked...

--- a/meta-ivi/conf/distro/poky-ivi-systemd.conf
+++ b/meta-ivi/conf/distro/poky-ivi-systemd.conf
@@ -44,4 +44,9 @@ SECURITY_LDFLAGS_pn-v86d = ""
 SECURITY_CFLAGS_pn-gettext = ""
 SECURITY_LDFLAGS_pn-gettext = ""
 
+# The unused variable was still there in dbus-1.12 (during certain #ifdef
+# conditions).  It seems to be fixed later, so this can be set to empty
+# string again.
+SECURITY_CFLAGS_pn-dbus = "-Wno-error=unused-variable"
+
 QEMU_TARGETS ?= "arm aarch64 i386 x86_64"

--- a/meta-ivi/recipes-core-ivi/dbus/dbus/build_fix-ubunt1604-qemux86.patch
+++ b/meta-ivi/recipes-core-ivi/dbus/dbus/build_fix-ubunt1604-qemux86.patch
@@ -1,24 +1,3 @@
---- dbus-1.10.10/dbus/dbus-sysdeps-unix.c.orig	2017-01-26 20:39:27.868191644 +0900
-+++ dbus-1.10.10/dbus/dbus-sysdeps-unix.c	2017-01-26 20:43:15.708197478 +0900
-@@ -4320,13 +4320,14 @@
-     {
-       for (;;)
-         {
--          struct dirent buf, *de;
--          int k, fd;
-+          struct dirent *de;
-+          int fd;
-           long l;
-           char *e = NULL;
- 
--          k = readdir_r (d, &buf, &de);
--          if (k != 0 || !de)
-+          errno = 0;
-+          de = readdir (d);
-+          if (!de && errno)
-             break;
- 
-           if (de->d_name[0] == '.')
 --- dbus-1.10.10/dbus/dbus-message.c.orig	2017-01-26 20:32:02.456180239 +0900
 +++ dbus-1.10.10/dbus/dbus-message.c	2017-01-26 20:32:27.288180875 +0900
 @@ -4128,7 +4128,7 @@
@@ -30,15 +9,3 @@
      {
        int *a = dbus_realloc(loader->unix_fds,
                              loader->max_message_unix_fds * sizeof(loader->unix_fds[0]));
---- dbus-1.10.10/dbus/dbus-shell.c.orig	2017-01-26 20:45:49.292201410 +0900
-+++ dbus-1.10.10/dbus/dbus-shell.c	2017-01-26 20:47:39.840204241 +0900
-@@ -419,7 +419,9 @@
-             {
-             case '\n':
-               if (!delimit_token (&current_token, &retval, error))
-+		{
- 		goto error;
-+		}
- 
- 		_dbus_string_free (&current_token);
- 

